meetings/2020-09-23.md

```diff
@@ -1,61 +1,167 @@
+
+    * [OpenTimelineIO GSoC student final report](https://karthikriyer.github.io/blog/2020/08/26/gsoc-20-final-report.html)
+
+    * Larry: it sounds like all 3 candidates "knocked it out of the park", the 3 projects would likely want to do this again. 2 projects didn’t get a “slot”, one project joined ASWF too late. Should we presume we want to do this again? Should we ask for more slots? What is the limit of how many students we could take on? We don’t have to decide this now. Joshua: would not want to have more than one at a time, time commitment would be too large. Wave: was there any communication between the students across channels? Brian: possibly not. Cary: Owen mentioned there was a forum (on the GSoC side) for students to communicate with each other. Some students on other projects mentioned that they didn’t get much mentorship. Wave: might be a good idea to provide some explicit way for ASWF interns to communicate with each other. Joshua: emphasis was to incorporate Kartic in OTIO communication channels, but would be helpful to connect the students together. Brian: would make sense to create a Slack channel for next year. Carol: from D&I perspective, improving this process and onboarding would help. Nick: Slack is a very useful channel for this.
+
+    * Larry: great summaries from everyone, we can pick up further discussion on the TAC mailing list / Slack channel. Let’s do it again next year, expand it so that every project can get one student, unless a project decides it wants more than one student (seems a consensus that 1 is about right).
+
+* Follow ups
+
+    * **Recommendation for VFX Platform Long-Term Support**
+
+        * Larry: 8 years into VFX Reference Platform
+
+        * Multiple "walls of compatibility"
+
+        * How far back should we support? CI currently only supports a couple of years back
+
+        * Cross project dependencies (OCIO / OIIO)
+
+        * Daniel: OpenVDB has clear policy of 3 years back (2018-2020), only project with clear policy. CI supports 2018 and on via containers.
+
+        * Availability of RedHat DTS for download
+
+        * Larry: does OpenVDB not issue patches for older versions? Many studios use Maya versions that still rely on gcc 4.8. Daniel: will ask OpenVDB folks explicitly.
```

Comment:
No, we don't. OpenVDB no longer supports GCC 4.8 as we use C++14 features. At ILM, we use an older version of OpenVDB (6.x) inside Maya 2018 that _does_ support GCC 4.8. The Maya plugin hasn't seen active development in a while though.

Reply:
Interesting! So you expect people who need OpenVDB in the Maya ecosystem to use old versions, and they're basically on their own, but current OpenVDB development doesn't test against that ecosystem and you no longer patch it if bugs are found? Do you get any push-back from users?

Reply:
I'll also add, since I neglected to mention at the meeting: OSL only continues to issue patches for one major version back routinely (i.e. the one "supported release"), will occasionally patch two major releases back for serious bugs if somebody requests it. But we make sure that both the current development branch and the currently maintained release family *build/test against* several years back worth of dependencies. Though we don't do this in a principled way -- I bet we do not test against a consistent number of years lookback among { C++, OpenEXR, OIIO, LLVM, etc. }.

So, there are really two questions we are trying to get at, and the answers may be different: (1) how many years back of platform/dependencies should we build and test against, and (2) how many years/releases back of our own software should we continue to patch/support?

Reply:
> Interesting! So you expect people who need OpenVDB in the Maya 2018 ecosystem to use old versions, and they're basically on their own, but current OpenVDB development doesn't test against that ecosystem and you no longer patch it if bugs are found? Do you get any push-back from users?

Correct. Maya is a bit of a special case though, it's essentially an orphaned plugin that hasn't been developed in 5+ years and we don't have CI for it. We very rarely get feature requests or bugs reported.

For VDB, as mentioned our policy for (1) is the last three years of the VFX Reference platform, this is explained in more detail here (https://github.com/AcademySoftwareFoundation/openvdb/blob/master/tsc/process/deprecation.md). For (2), we don't have a policy, but I would say we unofficially do the same as OSL, one major version back. We don't have scheduled builds to test previous versions though. In practice, the answer depends a little on whether it's a vendor or a studio/individual. It's understandable that a vendor sticking to the VFX Reference Platform might need a bugfix to the previous major release. For a studio/individual, we don't typically backport fixes (in fact I can't remember the last time we did so), we just fix the bug and encourage them to move up to a later version. I believe this hasn't been a problem because we've mostly introduced new features without many breaking API changes and we are meticulous about maintaining ABI - if your host application natively offers VDB 6.x, you can build VDB 7.x in ABI=6 mode for compatibility with the natively provided version. However, it's also possible that in lieu of us providing an official patch release, users are simply backporting fixes in their own fork. I know that's what SideFX has done in the past (https://github.com/sideeffects/openvdb_dev/tags).

Reply:
Thanks for the follow up here and in the TAC mailing list thread @danrbailey and @lgritz 

I'll merge this PR now so I guess we should move any further discussion there.